pkg/virt-api/validating-webhook/validating-webhook.go

```diff
@@ -608,29 +608,77 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 	}
 
 	if len(spec.Networks) > 0 && len(spec.Domain.Devices.Interfaces) > 0 {
+		multusExists := false
+		kuryrExists := false
+		genieExists := false
 		for idx, network := range spec.Networks {
-			if network.Pod == nil && network.Multus == nil {
+			if network.Pod == nil && network.Cni == nil {
 				causes = append(causes, metav1.StatusCause{
 					Type:    metav1.CauseTypeFieldValueRequired,
 					Message: fmt.Sprintf("should have a network type"),
 					Field:   field.Child("networks").Index(idx).Child("pod").String(),
 				})
 			}
 
-			if network.Pod != nil && network.Multus != nil {
+			if network.Pod != nil && network.Cni != nil {
 				causes = append(causes, metav1.StatusCause{
 					Type:    metav1.CauseTypeFieldValueRequired,
 					Message: fmt.Sprintf("should have only one network type"),
 					Field:   field.Child("networks").Index(idx).String(),
 				})
 			}
 
-			if network.Multus != nil && network.Multus.NetworkName == "" {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueRequired,
-					Message: fmt.Sprintf("multus network must have a networkName"),
-					Field:   field.Child("networks").Index(idx).Child("multus").String(),
-				})
+			if network.Cni != nil {
+				if network.Cni.NetworkName == "" {
+					causes = append(causes, metav1.StatusCause{
+						Type:    metav1.CauseTypeFieldValueRequired,
+						Message: fmt.Sprintf("CNI network must have a networkName"),
+						Field:   field.Child("networks").Index(idx).Child("cni").String(),
+					})
+				}
+
+				if network.Cni.Multus != nil {
+					if kuryrExists || genieExists {
```

Comment:
this logic is difficult to maintain as more network devices are added. Take a look at how the volume source handles ensuring only a single item is set.